test/http-mock/cheque-mock.ts

```diff
@@ -0,0 +1,85 @@
+import { createServer, Server } from 'http'
+
+/**
+ * Starts a mock HTTP server with predefined cheque responses.
+ *
+ * Can be run in standalone mode:
+ * ts-node test/http-mock/cheque-mock.ts run 1333
+ */
+export function createChequeMockHttpServer(port: number): Server {
+  const server = createServer((request, response) => {
+    response.writeHead(200, { 'Content-Type': 'application/json' })
+
+    if (request.url === '/health') {
+      response.end(JSON.stringify(health))
+    }
+
+    if (request.url === '/chequebook/cheque') {
+      response.end(JSON.stringify(lastCheques))
+    }
+
+    if (request.url === '/chequebook/balance') {
+      response.end(JSON.stringify(balance))
+    }
+
+    if (request.url === '/chequebook/cheque/1105536d0f270ecaa9e6e4347e687d1a1afbde7b534354dfd7050d66b3c0faad') {
+      response.end(JSON.stringify(cumulativeCheque1))
+    }
+
+    if (request.url === '/chequebook/cashout/1105536d0f270ecaa9e6e4347e687d1a1afbde7b534354dfd7050d66b3c0faad') {
+      response.end(JSON.stringify(lastCashoutCheque1))
+    }
+  })
+  server.listen(port)
+
+  return server
+}
+
+const health = { status: 'ok', version: '0.5.3-acbd0e2' }
+
+const lastCheques = {
+  lastcheques: [
+    {
+      peer: '1105536d0f270ecaa9e6e4347e687d1a1afbde7b534354dfd7050d66b3c0faad',
+      lastreceived: {
+        beneficiary: '0x1A66fd26F38819A96CA24404eA50A601Fa1Cb0c3',
+        chequebook: '0x7789F00C8cc0Aaa10D2fC50d9c7d74383fD9AA4d',
+        payout: 8944000000000,
+      },
+      lastsent: {
+        beneficiary: '0xe9dF78C7bd5162F93535657E7405143F2E9Cc260',
+        chequebook: '0x4f16B90D43aa1f9fd3c2fe17DACD9b21417A4134',
+        payout: 8293000000000,
+      },
+    },
+  ],
+}
+
+const cumulativeCheque1 = {
+  peer: '1105536d0f270ecaa9e6e4347e687d1a1afbde7b534354dfd7050d66b3c0faad',
+  lastreceived: {
+    beneficiary: '0x1A66fd26F38819A96CA24404eA50A601Fa1Cb0c3',
+    chequebook: '0x7789F00C8cc0Aaa10D2fC50d9c7d74383fD9AA4d',
+    payout: 8944000000000,
+  },
+  lastsent: {
+    beneficiary: '0xe9dF78C7bd5162F93535657E7405143F2E9Cc260',
+    chequebook: '0x4f16B90D43aa1f9fd3c2fe17DACD9b21417A4134',
+    payout: 8293000000000,
+  },
+}
+
+const lastCashoutCheque1 = {
+  peer: '1105536d0f270ecaa9e6e4347e687d1a1afbde7b534354dfd7050d66b3c0faad',
+  chequebook: '0x7789f00c8cc0aaa10d2fc50d9c7d74383fd9aa4d',
+  cumulativePayout: 0,
+  beneficiary: '0x1a66fd26f38819a96ca24404ea50a601fa1cb0c3',
+  transactionHash: '0x11df9811dc8caaa1ff4389503f2493a8c46b30c0a0b5f8aa54adbb965374c0ae',
+  result: { recipient: '0x4f16b90d43aa1f9fd3c2fe17dacd9b21417a4134', lastPayout: 0, bounced: false },
+}
+
+const balance = { totalBalance: 100026853000000000, availableBalance: 100018560000000000 }
+
+if (process.argv[2] === 'run') {
```

Comment:
Can you give me a little context about it?

Reply:
There is a JSDoc which I think explains this behaviour.

```
/**
 * Starts a mock HTTP server with predefined cheque responses.
 *
 * Can be run in standalone mode:
 * ts-node test/http-mock/cheque-mock.ts run 1333
 */
```

Is this sufficient?

Reply:
perfect : )